### DISCUSSION TITLE
Unable to adapt docker-image flow to git:from-image

### DISCUSSION THREAD

**Discussion by orov-io** [dokku/dokku#4835] [Q&A] [answered]
Hi all!

Currently in my company we use dokku to deploy docker images that have been previously uploaded to a dockerhub repository. The process in question means that we always end up deploying an image with the same tag in each of them. As an example, let's assume that in a development environment, we always deploy myapp:latest

To achieve this, whenever we have a new image to be deployed in that environment, we move the tag to the new image:

`latest points to the image with digest df4bc6bcede6 -> new image with digest a799dbe4aae7 ready -> latest points to the new image with digest a799dbe4aae7.`

This flow worked for us with the previous way of deployment based on the dokku tags command. The problem is that when migrating to git:from-image, we found this in the documentation:

> Triggering a build with the same arguments multiple times will result in Dokku exiting 0 early as there will be no changes detected.

and, indeed, even if the image that the latest tag refers to changes, since the arguments are the same on each attempt, dokku decides that there is nothing to deploy. Furthermore, doing ps:rebuild only results in dokku to re-deploy the image that was already deployed. I leave here a detail of the new flow:
```
# On dokku host:

$ dokku apps:create myapp
$ dokku git:from-image myapp $DOCKER_REGISTRY/myapp:latest // works, and latest is running in the server.

# On CI pipeline

// first we move the latest tag to the image we want to deploy and:
$ ssh $DOKKU_REMOTE git:from-image myapp $DOCKER_REGISTRY/myapp:latest // Dokku doesn't found changes and do nothing
```
Any idea how to migrate this workflow to the new dokku form? Are we using the new functionality wrong and any extra steps need to be done?  We think that ideally dokku should check the image digest to decide if it has to download a new image or not, instead of relying on the tag.

**Comment by josegonzalez** [comment, marked as answer]
I've [updated the documentation](https://github.com/dokku/dokku/pull/4840), but you should instead use the image when reusing tags.

So in the above case, your CI pipeline should get the digest:

```shell
docker inspect --format='{{index .RepoDigests 0}}' $DOCKER_REGISTRY/myapp:latest
```

And then use the output when calling git:from-image:

```shell
ssh $DOKKU_REMOTE git:from-image myapp $DOCKER_REGISTRY/myapp@sha256:whateverthatshais
```

I believe the docker inspect command above will also include the repo name in it, but you'll need to verify that.

**Reply by orov-io** [reply]
Hi @josegonzalez,

solution works for us, so thanks a lot for your help. Output of the docker command that you say includes the repo name, as:

> $DOCKER_REGISTRY/myapp@sha256:0cdf8ae90d4fe1af5fde60c229b2c08fc4b08359fc6852f35a9625fe7e450f7b

It may be interesting to point in the documentation that before the docker inspect command, the selected image must be pulled, because it is possible that a developer has it available on his local machine, but when migrating the flow to a CI tool, this is not the case and docker will exit with an error because it does not find the image:

```
docker pull $DOCKER_REGISTRY/myapp:latest
docker inspect --format='{{index .RepoDigests 0}}' $DOCKER_REGISTRY/myapp:latest
```
